examples/crowdfunding/test/crowdfundingTest.js

```diff
@@ -0,0 +1,252 @@
+  getProgram,
+  intToBigEndian,
+  SignType,
+  stringToBytes,
+  TransactionType
+} from '@algorand-builder/algob';
+import { Runtime, StoreAccountImpl } from '@algorand-builder/algorand-js/build/src/index';
+import { assert } from 'chai';
+
+import { getAcc } from './common';
+
+const initialDonorBalance = 60000000;
+const initialCreatorBalance = 10000;
+const goal = 7000000;
+
+describe('Crowdfunding Tests', function () {
+  let creatorAccount = new StoreAccountImpl(initialCreatorBalance);
+  let escrowAccount = new StoreAccountImpl(0);
+  let donorAccount = new StoreAccountImpl(initialDonorBalance);
+
+  let runtime;
+  let program;
+  let flags;
+  let applicationId;
+  this.beforeAll(async function () {
+    runtime = new Runtime([creatorAccount, escrowAccount, donorAccount]);
+    program = getProgram('crowdFundApproval.teal');
+
+    flags = {
+      sender: creatorAccount.account,
+      localInts: 1,
+      localBytes: 0,
+      globalInts: 5,
+      globalBytes: 3
+    };
+  });
+
+  // update account state
+  function syncAccounts () {
+    creatorAccount = getAcc(runtime, creatorAccount);
+    escrowAccount = getAcc(runtime, escrowAccount);
+    donorAccount = getAcc(runtime, donorAccount);
+  }
+
+  it('should create crowdfunding application with correct global states', async () => {
+    const creationFlags = Object.assign({}, flags);
+
+    // Get begin date to pass in
+    const beginDate = new Date();
+    beginDate.setSeconds(beginDate.getSeconds() + 2);
+
+    // Get end date to pass in
+    const endDate = new Date();
+    endDate.setSeconds(endDate.getSeconds() + 12000);
+
+    // Get fund close date to pass in
+    const fundCloseDate = new Date();
+    fundCloseDate.setSeconds(fundCloseDate.getSeconds() + 120000);
+
+    const appArgs = [
+      intToBigEndian(beginDate.getTime()),
+      intToBigEndian(endDate.getTime()),
+      intToBigEndian(goal),
+      addressToBytes(creatorAccount.address),
+      intToBigEndian(fundCloseDate.getTime())
+    ];
+
+    const appId = await runtime.addApp({ ...creationFlags, appArgs: appArgs }, {}, program);
+    applicationId = appId;
+    const creatorPk = addressToBytes(creatorAccount.address);
+
+    // verify global state
+    assert.isDefined(appId);
+    assert.deepEqual(runtime.getGlobalState(appId, stringToBytes('Creator')), creatorPk);
+    assert.deepEqual(runtime.getGlobalState(appId, stringToBytes('StartDate')), BigInt(beginDate.getTime()));
+    assert.deepEqual(runtime.getGlobalState(appId, stringToBytes('EndDate')), BigInt(endDate.getTime()));
+    assert.deepEqual(runtime.getGlobalState(appId, stringToBytes('Goal')), BigInt(7000000));
+    assert.deepEqual(runtime.getGlobalState(appId, stringToBytes('Receiver')), creatorPk);
+    assert.deepEqual(runtime.getGlobalState(appId, stringToBytes('Total')), BigInt(0));
+  });
+
+  it('should update application with correct escrow account address', async () => {
+    const appArgs = [addressToBytes(escrowAccount.address)];
```

Comment:
Please rename the function `addressToBytes` to `addressToPk`/`decodeAddress`. (It's confusing when kept inline with `stringToBytes`, `base64ToBytes`)

Reply:
seeing from user perspective, i think we should keep it because it maybe confusing to user when he is trying to pass `appArgs`. For developers we can update the doc string

Reply:
Fair enough. Can you add a doc string here as well?

Reply:
can you please suggest something @ratik21 ?

Reply:
How about:
`// addressToBytes decodes algorand address from string to Uint8Array (as public key)`

Reply:
works.